homeassistant/components/climate/ambiclimate.py

```diff
@@ -0,0 +1,271 @@
+                       link_name=CONFIGURATOR_LINK_NAME,
+                       link_url=oauth.get_authorize_url(),
+                       description=CONFIGURATOR_DESCRIPTION,
+                       submit_caption=CONFIGURATOR_SUBMIT_CAPTION)
+            hass.data[AMBICLIMATE_DATA] = data
+
+        hass.async_add_job(_call_request_config)
+        return
+
+    if hass.data.get(AMBICLIMATE_DATA):
+        def _call_request_config_done():
+            configurator = hass.components.configurator
+            configurator.request_done(hass.data.get(AMBICLIMATE_DATA))
+            del hass.data[AMBICLIMATE_DATA]
+
+        hass.async_add_job(_call_request_config_done)
+
+    websession = async_get_clientsession(hass)
+    data_connection = ambiclimate.AmbiclimateConnection(oauth,
+                                                        token_info=token_info,
+                                                        websession=websession)
+
+    if await data_connection.refresh_access_token():
+        await store.async_save(data_connection.token_info)
+
+    if not await data_connection.find_devices():
+        return
+
+    devs = []
+    for heater in data_connection.get_devices():
+        await heater.update_device_info()
+        devs.append(Ambiclimate(heater, store))
+
+    async_add_entities(devs, True)
+
+    async def send_comfort_feedback(service):
+        """Send comfort feedback."""
+        device_name = service.data.get(ATTR_NAME)
+        device = data_connection.find_device_by_room_name(device_name)
+        if device:
+            await device.set_comfort_feedback(service.data.get(ATTR_VALUE))
+
+    hass.services.async_register(DOMAIN,
+                                 SERVICE_COMFORT_FEEDBACK,
+                                 send_comfort_feedback,
+                                 schema=SEND_COMFORT_FEEDBACK_SCHEMA)
+
+    async def set_comfort_mode(service):
+        """Set comfort mode."""
+        device_name = service.data.get(ATTR_NAME)
+        device = data_connection.find_device_by_room_name(device_name)
+        if device:
+            await device.set_comfort_mode()
+
+    hass.services.async_register(DOMAIN,
+                                 SERVICE_COMFORT_MODE,
+                                 set_comfort_mode,
+                                 schema=SET_COMFORT_MODE_SCHEMA)
+
+    async def set_temperature_mode(service):
+        """Set temperature mode."""
+        device_name = service.data.get(ATTR_NAME)
+        device = data_connection.find_device_by_room_name(device_name)
+        if device:
+            await device.set_temperature_mode(service.data.get(ATTR_VALUE))
+
+    hass.services.async_register(DOMAIN,
+                                 SERVICE_TEMPERATURE_MODE,
+                                 set_temperature_mode,
+                                 schema=SET_TEMPERATURE_MODE_SCHEMA)
+
+
+class AmbiclimateAuthCallbackView(HomeAssistantView):
```

Comment:
Use the `webhook` component instead of registering a view.

Reply:
I do not think it is possible to use the webhook component here, since the oauth authorization requires a fix callback URL that is associated with the client id and client secret.
(It also needs to support get requests)